Entities/Common/Controls/HighlightItems.as

```diff
@@ -0,0 +1,63 @@
+#define CLIENT_ONLY
+
+//Items to highlight.
+const string[] classes = {"builder", "knight", "archer"};
+const string[][] highlight_items = {
+/* 0 */	{"mat_stone", "mat_wood", "mat_gold"}, //builder
+/* 1 */	{"mat_bombs", "mat_waterbombs"}, //knight
+/* 2 */	{"mat_firearrows", "mat_waterarrows", "mat_bombarrows"} //archer
+};
+
+//If button is pressed.
+bool do_highlight = false;
+
+//Disable highlighting for items, if their luminance is less than this variable.
+const uint hide_luminance_level = 30;
+
+void onTick(CSprite@ sprite)
+{
+	do_highlight = sprite.getBlob().isKeyPressed(key_pickup);
+}
+
+void onRender(CSprite@ sprite)
+{
+	if (do_highlight)
+	{
+		//Highlight items.
+		CCamera@ camera = getCamera();
+		CMap@ map = getMap();
+		if (camera is null || map is null) return;
+
+		CBlob@[] blobs;
+		if (getBlobs(@blobs))
+		{
+			for (uint i = 0; i < blobs.length; i++)
+			{
+				CBlob@ blob = blobs[i];
+				//Highlight.
+				int class_index = classes.find(sprite.getBlob().getConfig());
+				if (shouldHighlightBlob(class_index, blob) && !blob.isInInventory())
+				{
+					float luminance = map.getColorLight(blob.getPosition()).getLuminance();
+					//Don't highlight items in dark caves and places.
+					if (luminance >= hide_luminance_level)
+					{
+						//Highlight like any normal pickup.
+						blob.RenderForHUD(Vec2f_zero, 0.0f, SColor(255,255,255,255), RenderStyle::normal);
+						//But do a beautiful fading effect.
+						uint brightness_level = Maths::Abs(Maths::Sin(getGameTime() / 20.0f) * 180);
```

Comment:
I would try to keep the period in seconds separate, as in, make the multiplication by 2pi explicit